Formula/gradle@6.rb

```diff
@@ -1,10 +1,15 @@
 class GradleAT6 < Formula
   desc "Open-source build automation tool based on the Groovy and Kotlin DSL"
   homepage "https://www.gradle.org/"
-  url "https://services.gradle.org/distributions/gradle-6.9.2-all.zip"
-  sha256 "8bde5c859a3ddf5d127ac77465fc24fa8a831d3d8d49e8248548f2cb87485ef1"
+  url "https://services.gradle.org/distributions/gradle-6.9.3-all.zip"
+  sha256 "91bbf438ec3f18c9273392972e5627bb9965ef88b262729d1b243d7db5cdd1b8"
   license "Apache-2.0"
 
+  livecheck do
+    url "https://gradle.org/releases/"
+    regex(/href=.*?gradle[._-]v?(6(?:\.\d+)+)-all\.(?:zip|t)/i)
```

Comment:
AFAIK, Gradle artifact is actually always zip, maybe we can simplify with `.zip` Also we are trying to avoid using `?:`.

```suggestion
    regex(/href=.*?gradle[._-]v?(6(?:\.\d+)+)-all\.zip/i)
```

Reply:
I'm generally fine with this either way. They seem to have consistently used zip files since 2009 (the earliest release on that page), so `\.zip` is fine here. However, if we really want to match either zip or tarball, the prevailing pattern is `(?:t|zip)`.

> Also we are trying to avoid using `?:`.

`(?:...)` simply indicates a non-capturing group. That is to say, anything in parentheses is captured by default, so we use non-capturing groups for parts of the regex that we don't want to capture. In almost all cases, we only want a capturing group around the version text and any other groups should be non-capturing (as mentioned in the related part of the [`brew livecheck` documentation](https://docs.brew.sh/Brew-Livecheck#regex-guidelines)). The exception are `strategy` blocks that work with more than one capture group, which isn't uncommon in homebrew/cask.

You may be mixing this up with lookarounds (e.g., `(?=...)`, `(?<=...)`, `(?!=...)`, `(?<!...)`), which we do try to avoid (per our previous conversation).

Reply:
I see, that works then :)

Reply:
yeah, the artifact is always `.zip` ever since I am using it.

Reply:
I copied livecheck from `gradle` formula. Not sure how long livecheck will be used as Gradle 8 is close to release (RC2). I didn't find much info on EOL for older Gradle release trains.